device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -106,43 +102,19 @@ void open(boolean withRetry) throws IOException
             {
                 this.transport.open(withRetry);
             }
-            catch (DeviceClientException e)
+            catch (TransportException e)
```

Comment:
Yes

Reply:
Would the TransportException become the inner exception though? In that case, it isn't fully hidden from the user, just not the parent-level type they need to catch.

Reply:
It would be the inner exception, yes. Users probably won't catch or even check the inner exception since an "AmqpConnectionForcedException" means nothing to them, but they will be there for debugging purposes when a user complains about the SDK throwing a particular IotHubClientException